decidim/decidim
# Implement in person vote

#### :tophat: What? Why?
This PR allow polling officers to register in person votes in the system. With an important refactor of the vote flow implementation, it implements several features:
* It shows the ballot style for the identified participant, as described in #7125.
* It marks that the participant has voted, as stated in #7126.
* It completes #7828 to fulfill all the acceptance criterias of #7124, checking if the participant has voted online or in person before.
* It modifies the online voting process, preventing participants to vote online after voting in person.
* Finally, it fixes some issues discovered during its development:
  * the autojump feature was not working for any form.
  * the locale setting was being lost when clicking on the "Trustees" link on the admin menu of a Voting space.
  * the admin ballot styles edition page was not very usable, as the checkboxes for the questions were not correctly related to their labels.
  * the validation of the birthdate was failing when filled with non-numerical data.

#### :pushpin: Related Issues
- Fixes #7121, #7124, #7125 and #7126

#### Testing
* Configure a Voting with polling stations, polling officers, a census and an election created on the Bulletin Board. Advance with the election setup until having it in the vote status.
* Login with a polling officer user, go to their polling officer zone (in the "My account" page) and go to identify a voter.
* Complete the form with the data of a voter included in the census and verify their data.
* Check that the questions shown match the ballot style assigned to the voter and complete the voting process.
* Try to do the same process again with the same data and check that the voter can't vote again.
* Try to vote online with the voter data and check that this can't be done.

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](https://github.com/decidim/decidim/blob/develop/CONTRIBUTING.adoc) to this repository.


### :camera: Screenshots

https://user-images.githubusercontent.com/453545/115590549-60c01e00-a2d1-11eb-9632-02067b7e23bb.mov

https://user-images.githubusercontent.com/453545/115590976-e5ab3780-a2d1-11eb-9c18-f55b8f9f708c.mov




:hearts: Thank you!
